File: backend/olympic_it_project/src/main/java/org/example/olympic_ot_project/dto/classes/CreateClassRequest.java
package org.example.olympic_ot_project.dto.classes;

import jakarta.validation.constraints.NotBlank;
import lombok.Data;

@Data
public class CreateClassRequest {
    @NotBlank(message = "Tên lớp không được để trống")
    private String className;
    @NotBlank(message = "Năm học không được để trống")
    private Integer academicYearId;
}

File: frontend/olympic_it_project/olympic_it_project/lib/dto/admin_manager/classes/create_class_request.dart
class CreateClassRequest {
  final String className;
  final int academicYearId;

  CreateClassRequest({
    required this.className,
    required this.academicYearId,
  });

  Map<String, dynamic> toJson() {
    return {
      "className": className,
      "academicYearId": academicYearId,
    };
  }
}